lib/asset.js

```diff
@@ -87,13 +126,108 @@ class Asset {
         $elementToReplace.replaceWith(this.getAssetTag());
     }
 
-    isValid () {
-        // TODO: Validate metadata fields
-        return !this.failed;
+    addCustomFailure (message) {
+        this._customFailures.push(message);
+    }
+
+    // FIXME use it
+    validateCanonicalURI (value) {
+        const errors = [];
+
+        if (validator.isEmpty(_coerce(value))) {
+            errors.push('Can\'t be empty');
+        }
+
+        // FIXME can also be validator.isDataURI ?
+        if (!validator.isURL(_coerce(value))) {
+            errors.push('Must be a valid URL');
+        }
+
+        return errors;
+    }
+
+    validateContentType (value) {
```

Comment:
Yes, the current exception is videos, they are downloaded in SOMA. The same will be true for audio, when we add support for them. Maybe we could generalize to "if something doesn't have a data file, it will be downloaded later in the workflow, and content type will be filled after that."